tools/add_font.py

```diff
@@ -23,6 +55,30 @@
                      ' for a -ext subset.')
 
 
+def family_url(name):
+  url_prefix = 'http://fonts.google.com/specimen/'
+  dl_name = name.replace(' ', '+')
+  return url_prefix + dl_name
+
+
+def family_exists(name):
+  """Check if a font family exists on fonts.google.com"""
+  url = family_url(name)
+  request = requests.get(url)
+  if request.status_code == 200:
+      return True
+  return False
+
+
+def _GetFamilyMetaData(name):
+  """Get family information from the frontend GF api"""
+  api_prefix = 'http://fonts.google.com/metadata/fonts/'
```

Comment:
I believe it can all be obtained from the public developer API, yes, eg

```json
  {
   "kind": "webfonts#webfont",
   "family": "ABeeZee",
   "category": "sans-serif",
   "variants": [
    "regular",
    "italic"
   ],
   "subsets": [
    "latin"
   ],
   "version": "v10",
   "lastModified": "2017-05-11",
   "files": {
    "regular": "http://fonts.gstatic.com/s/abeezee/v10/mE5BOuZKGln_Ex0uYKpIaw.ttf",
    "italic": "http://fonts.gstatic.com/s/abeezee/v10/kpplLynmYgP0YtlJA3atRw.ttf"
   }
```